contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -2088,6 +2089,172 @@ numeric_radians(PG_FUNCTION_ARGS)
 	PG_RETURN_NUMERIC(result);
 }
 
+Datum
+parsename(PG_FUNCTION_ARGS)
+{
+	text *object_name = PG_GETARG_TEXT_PP(0);
+	int object_piece = PG_GETARG_INT32(1);
+	char *object_name_str = text_to_cstring(object_name);
+	int len = strlen(object_name_str);
+	char object_parts[129 * 5] = {0};
```

Comment:
As Di and Zhibai mentioned in previous comments, we don't need this buffer. 
The target here is to separate out four parts from the input, so at most we only need to manage four pointers pointing to the start of each part, and record the length of each part. 
In fact, since we know which part needs to be returned, one pointer should be enough.

Reply:
i feel that we may at most need 256 B for one part. 
Please be noticed that according to the doc, one part is up to 256B 
I am not sure why it is 129 x 5 here.

Reply:
array called "object_parts" that is 129 * 5 in size, meaning it can hold up to 5 elements, each of which is a null-terminated string of maximum length 128 characters. This array is used to store the different parts of an object name that is passed in as an argument to the function